state/protocol/badger/validity.go

```diff
@@ -135,21 +135,21 @@ func isValidRootSnapshot(snap protocol.Snapshot, verifyResultID bool) error {
 		return fmt.Errorf("could not latest sealed result: %w", err)
 	}
 
-	if len(segment) == 0 {
+	if len(segment.Blocks) == 0 {
 		return fmt.Errorf("invalid empty sealing segment")
 	}
 	// TAIL <- ... <- HEAD
-	head := segment[len(segment)-1] // reference block of the snapshot
-	tail := segment[0]              // last sealed block
-	headID := head.ID()
-	tailID := tail.ID()
+	highest := segment.Highest() // reference block of the snapshot
+	lowest := segment.Lowest()   // last sealed block
+	highestID := highest.ID()
+	lowestID := lowest.ID()
 
-	if result.BlockID != tailID {
-		return fmt.Errorf("root execution result for wrong block (%x != %x)", result.BlockID, tail.ID())
+	if result.BlockID != lowestID {
+		return fmt.Errorf("root execution result for wrong block (%v != %v)", result.BlockID, lowest)
```

Comment:
This is functionally the same, because we implement a handler for `%v` for the `Identifier` type, but I think it is better to use the explicit formatting verb as much as possible so it is easy to see at a glance what the resulting string will look like.

Reply:
The linter was failing , thats why I made these to %v updates. Did we change something with the linter?

Reply:
Hmm, what lint error are you getting?

Reply:
typo on my end